core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java

```diff
@@ -521,44 +526,90 @@ public static final class Metrics {
     public static final Counter WORKER_ACTIVE_CLIENTS =
         MetricsSystem.counter(MetricKey.WORKER_ACTIVE_CLIENTS.getName());
 
+    /**
+     * Update mMetricCache when find it exceed certain interval.
+     * @param cache the BlockMetaMetricCache
+     */
+    public static void maybeUpdateMetrics(BlockMetaMetricCache cache) {
```

Comment:
You can utilize the `com.codahale.metrics.CachedGauge` class provided by the metrics system, so that you don't have to repeat the cache-or-update logic.